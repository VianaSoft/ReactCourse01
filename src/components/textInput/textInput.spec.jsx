/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react';

import { TextInput } from '.';
import userEvent from '@testing-library/user-event';

describe('<TextInput />', () => {
  it('shoud call handleChange function on each key pressed', () => {
    const fn = jest.fn();
    render(<TextInput handleChange={fn} searchValue={'Gold'} />);
    const input = screen.getByPlaceholderText(/Enter your search/i);
    const value = 'Ultimate';

    userEvent.type(input, value);

    expect(input.value).toBe('Gold');
    expect(fn).toHaveBeenCalledTimes(value.length);
  });

  it('shoud have a value of searchValue', () => {
    const fn = jest.fn();
    render(<TextInput handleChange={fn} searchValue={'Gold'} />);

    const input = screen.getByPlaceholderText(/Enter your search/i);
    expect(input).toBeInTheDocument();
    expect(input.value).toBe('Gold');
    expect(input.type).toBe('search');
  });

  it('Should match snapshot', () => {
    const fn = jest.fn();
    const { container } = render(<TextInput handleChange={fn} searchValue={'Gold'} />);

    expect(container).toMatchSnapshot();
  });
});
