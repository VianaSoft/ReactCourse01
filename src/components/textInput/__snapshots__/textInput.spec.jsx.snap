// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> Should match snapshot 1`] = `
<div>
  <input
    placeholder="Enter your search"
    type="search"
    value="Gold"
  />
</div>
`;
